File: zul/src/org/zkoss/zul/api/Toolbar.java
/* Toolbar.java

{{IS_NOTE
	Purpose:
		
	Description:
		
	History:
		Tue Oct 22 14:45:31     2008, Created by Flyworld
}}IS_NOTE

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under GPL Version 3.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
 */
package org.zkoss.zul.api;

import org.zkoss.zk.ui.WrongValueException;

/**
 * A toolbar.
 * 
 * <p>
 * Mold:
 * <ol>
 * <li>default</li>
 * <li>panel: since 3.5.0, this mold is used for {@link Panel} component as its
 * foot toolbar.</li>
 * </ol>
 * <p>
 * Default {@link #getZclass}: z-toolbar, if {@link #getMold()} is panel,
 * z-toolbar-panel is assumed.(since 3.5.0)
 * 
 * @author tomyeh
 * @since 3.5.2
 */
public interface Toolbar extends org.zkoss.zul.impl.api.XulElement {

	/**
	 * Returns the alignment of any children added to this toolbar. Valid values
	 * are "start", "end" and "center".
	 * <p>
	 * Default: "start"
	 * 
	 */
	public String getAlign();

	/**
	 * Sets the alignment of any children added to this toolbar. Valid values
	 * are "start", "end" and "center".
	 * <p>
	 * Default: "start", if null, "start" is assumed.
	 * 
	 */
	public void setAlign(String align);

	/**
	 * Returns the orient.
	 * <p>
	 * Default: "horizontal".
	 */
	public String getOrient();

	/**
	 * Sets the orient.
	 * 
	 * @param orient
	 *            either "horizontal" or "vertical".
	 */
	public void setOrient(String orient) throws WrongValueException;

}

File: zul/src/main/resources/web/js/zul/wgt/mold/toolbar.js
/* toolbar.js

	Purpose:

	Description:

	History:
		Sat Dec 24 12:58:43     2008, Created by Flyworld

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 2.1 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
function toolbar$mold$(out) {
	var uuid = this.uuid;

	out.push('<div ', this.domAttrs_(), '><div id="', uuid, '-cave"',
		' class="', this.$s('content'), ' ', this.$s(this.getAlign()), '" role="none" >');

	this._tabindexTargetChild = undefined;
	for (var w = this.firstChild; w; w = w.nextSibling) {
		w.redraw(out);
	}
	out.push('</div>');

	if (this.isOverflowPopup()) {
		out.push('<i id="', uuid, '-overflowpopup-button"',
			' class="', zUtl.encodeXML(this._getOverflowPopupBtnClass()), '" aria-hidden="true"></i>');
		out.push('<div id="', uuid, '-pp" class="', this.$s('popup'), ' ', this.$s('popup-close'), '"></div>');
	}
	out.push('<div class="z-clear" aria-hidden="true"></div></div>');
}
